t/lib/OpenQA/Test/CheckGitStatus.pm

```diff
@@ -16,19 +16,22 @@ if ($CHECK_GIT_STATUS) {
 }
 
 sub check_status {
-    local $?;
-    chdir $cwd;
-    my $git = File::Which::which('git');
-    return unless $git;
-    my $cmd = 'git rev-parse --git-dir';
-    my $out = qx{$cmd};
-    return if $? != 0;
-    $cmd = 'git status --porcelain=v1 2>&1';
-    my @lines = qx{$cmd};
-    die "Problem running git:\n" . join '', @lines if $? != 0;
+    my @lines;
+    {
+        local $?;
+        chdir $cwd;
+        my $git = File::Which::which('git');
+        return unless $git;
+        my $cmd = 'git rev-parse --git-dir';
+        my $out = qx{$cmd};
+        return if $? != 0;
+        $cmd = 'git status --porcelain=v1 2>&1';
+        @lines = qx{$cmd};
+        die "Problem running git:\n" . join '', @lines if $? != 0;
+    }
     if (@lines > 0) {
         Test::More::diag("Error: modified or untracked files\n" . join '', @lines);
-        exit 1;
+        $? = 1;
```

Comment:
I need to ask the obvious, #1873 was passing CI - how do I know this will work? Can we check this anywhere?

Reply:
@perlpunk was able to reproduce the problem locally so I assume she's tested it locally.

Reply:
Yes, I tested both cases locally now by
* adding a `fail()` to 13-os-utils.t to see if `prove` really fails if a test fails
* adding an untracked file to see if the status check still works